protostar/cheatable_starknet/controllers/contracts.py

```diff
@@ -103,6 +131,12 @@ async def _transform_calldata_to_cairo_data(
     ) -> CairoData:
         if isinstance(calldata, collections.Mapping):
             contract_class = await self.cheatable_state.get_contract_class(class_hash)
+
+            if isinstance(contract_class, CompiledClass):
+                # TODO add handling
+                raise ValueError("Cair1 contracts do not support data transformation.")
```

Comment:
Do we need better handling here?

Reply:
It depends it will be raised as Unexpected exception by protostar
If you want exception to be pretty printed you have to raise CheatcodeException

Reply:
But it should be catched and re-raised from HintLocal not from Controller for consistency

Reply:
@Arcticae

Reply:
That's right 👍

Reply:
It's not used in Declare cheatcode, so perhaps this should be addressed when updating relevant cheatcodes (deploy_prepared and prepare I believe)?